packages/presentation/lib/screens/login/login_screen.dart

```diff
@@ -0,0 +1,205 @@
+                                style:
+                                    LoginScreenStyles.forgotPasswordLabelStyle,
+                              ),
+                            ),
+                          ),
+                          SizedBox(
+                            width: double.infinity,
+                            child: ElevatedButton(
+                              style: ElevatedButton.styleFrom(
+                                backgroundColor: AppColors.red,
+                              ),
+                              onPressed: bloc.onLogin,
+                              child: Padding(
+                                padding: const EdgeInsets.symmetric(
+                                  vertical: AppSizes.size15,
+                                ),
+                                child: Text(
+                                  appLocalizations.loginButtonLabel,
+                                  style: LoginScreenStyles.loginButtonTextStyle,
+                                ),
+                              ),
+                            ),
+                          ),
+                        ],
+                      ),
+                      Padding(
+                        padding: const EdgeInsets.only(bottom: AppSizes.size35),
+                        child: Row(
+                          mainAxisAlignment: MainAxisAlignment.center,
+                          children: [
+                            const AuthIconButtonWidget(
+                              AssetsImagesPaths.twitterIconPath,
+                              color: LoginScreenColors.twitterColor,
+                              onPressAction: null,
+                            ),
+                            const SizedBox(width: AppSizes.size24),
+                            AuthIconButtonWidget(
+                              AssetsImagesPaths.facebookIconPath,
+                              color: LoginScreenColors.facebookColor,
+                              onPressAction: bloc.authByFacebook,
+                            ),
+                            const SizedBox(width: AppSizes.size24),
+                            AuthIconButtonWidget(
+                              AssetsImagesPaths.googleIconPath,
+                              color: LoginScreenColors.googleColor,
+                              onPressAction: bloc.authByGoogle,
+                            ),
+                          ],
+                        ),
+                      ),
+                    ],
+                  ),
+                ),
+              );
+            } else {
+              return Container();
+            }
+          }),
+    );
+  }
+}
+
+TextEditingController _createTextController(String value) {
+  final textEditingController = TextEditingController(text: value);
+  textEditingController.selection = TextSelection(
+    baseOffset: value.length,
+    extentOffset: value.length,
+  );
+
+  return textEditingController;
+}
```

Comment:
why do u need it?

Reply:
To move cursor after last entered key without it entering result in field will be in reverse order

Reply:
let's put the controller into the bloc and trigger an update from it. like controller.text = text

Reply:
Fixed.